### PR TITLE
[FLINK-36162] Remove flinkStateSnapshotReference and namespace from FlinkStateSnapshot jobReference

### PR DESCRIPTION
**Pull request by gyfora**
## Change log

  - *Remove the namespace from snapshot job reference*
  - *Remove job.flinkStateSnapshotReference*
  - *Rename upgradeSavepointReference to upgradeSavepointPath*

## Verifying this change

Unit tests and e2es already cover this

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: yes
  - Core observer or reconciler logic that is regularly executed: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? docs updated

### CODE REVIEW

**Comment by gyfora** [comment]
cc @anupamaggarwal

File: docs/content/docs/custom-resource/reference.md [hunk 1]
```diff
@@ -188,11 +176,10 @@ This serves as a full reference for FlinkDeployment and FlinkSessionJob custom r
...
 | checkpointTriggerNonce | java.lang.Long | Nonce used to manually trigger checkpoint for the running job. In order to trigger a checkpoint, change the number to a different non-null value. |
 | upgradeMode | org.apache.flink.kubernetes.operator.api.spec.UpgradeMode | Upgrade mode of the Flink job. |
 | allowNonRestoredState | java.lang.Boolean | Allow checkpoint state that cannot be mapped to any job vertex in tasks. |
-| savepointRedeployNonce | java.lang.Long | Nonce used to trigger a full redeployment of the job from the savepoint path specified in initialSavepointPath or the path/FlinkStateSnapshot reference in flinkStateSnapshotReference. In order to trigger redeployment, change the number to a different non-null value. Rollback is not possible after redeployment. |
+| savepointRedeployNonce | java.lang.Long | Nonce used to trigger a full redeployment of the job from the savepoint path specified in initialSavepointPath.In order to trigger redeployment, change the number to a different non-null value. Rollback is not possible after redeployment. |
```
Comment (ferenc-csaky) [inline docs/content/docs/custom-resource/reference.md:182]: nit: "initialSavepointPath.In" -> "\`initialSavepointPath`. In"

File: flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/spec/JobSpec.java [hunk 2]
```diff
@@ -100,9 +92,8 @@ public class JobSpec implements Diffable<JobSpec> {
 
     /**
      * Nonce used to trigger a full redeployment of the job from the savepoint path specified in
-     * initialSavepointPath or the path/FlinkStateSnapshot reference in flinkStateSnapshotReference.
-     * In order to trigger redeployment, change the number to a different non-null value. Rollback
-     * is not possible after redeployment.
+     * initialSavepointPath.In order to trigger redeployment, change the number to a different
```
Comment (ferenc-csaky) [inline flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/spec/JobSpec.java:95]: nit: "initialSavepointPath.In order" -> "initialSavepointPath. In order"

**Review by ferenc-csaky — APPROVED** [review]
I think this is a reasonable change according to the explanation on the relevant jira.

Went through the changes, with the modified tests this LGTM. Added 2 nitty comments for a doc typo.

File: flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/spec/JobSpec.java (hunk 2 above)
Comment (gyfora) [reply flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/spec/JobSpec.java:95]: good catch

**Pull request merged**